src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -376,18 +345,17 @@ public bool TryRemove(KeyValuePair<TKey, TValue> item)
         /// <param name="value">The variable into which the removed value, if found, is stored.</param>
         /// <param name="matchValue">Whether removal of the key is conditional on its value.</param>
         /// <param name="oldValue">The conditional value to compare against if <paramref name="matchValue"/> is true</param>
-        /// <returns></returns>
         private bool TryRemoveInternal(TKey key, [MaybeNullWhen(false)] out TValue value, bool matchValue, [AllowNull] TValue oldValue)
         {
-            int hashcode = _comparer.GetHashCode(key);
+            IEqualityComparer<TKey>? comparer = _comparer;
+            uint hashcode = (uint)(comparer is null ? key.GetHashCode() : comparer.GetHashCode(key));
```

Comment:
I see more than a dozen of `int`-to-`uint` conversions for hash codes in these changes.  Is it possible to continue treating hash codes as `int` everywhere except for the modulo operation?  That would improve readability.